todoapp/app/src/test/java/com/example/android/architecture/blueprints/todoapp/statistics/StatisticsViewModelTest.java

```diff
@@ -0,0 +1,123 @@
+        mTasks = Lists.newArrayList(
+                new Task("Title1", "Description1"),
+                new Task("Title2", "Description2", true),
+                new Task("Title3", "Description3", true));
+        mProgressIndicatorTestSubscriber = new TestSubscriber<>();
+        mStatisticsTestSubscriber = new TestSubscriber<>();
+    }
+
+    @Test
+    public void getProgressIndicator_emitsFalseInitially() {
+        // When subscribing to the progress indicator
+        mViewModel.getProgressIndicator().subscribe(mProgressIndicatorTestSubscriber);
+
+        // One value: false, is emitted
+        mProgressIndicatorTestSubscriber.assertValue(false);
+    }
+
+    @Test
+    public void getStatistics_withTasks_returnsCorrectData() {
+        //Given a list of tasks in the repository
+        when(mTasksRepository.getTasks()).thenReturn(Observable.just(mTasks));
+
+        //When subscribing to the statistics stream
+        mViewModel.getStatistics().subscribe(mStatisticsTestSubscriber);
+
+        //The correct pair is returned
+        Pair<Integer, Integer> result = mStatisticsTestSubscriber.getOnNextEvents().get(0);
+        assertThat(result.first, is(1));
+        assertThat(result.second, is(2));
+    }
+
+    @Test
+    public void getStatistics_withNoTasks_returnsCorrectData() {
+        //Given a list of tasks in the repository
+        when(mTasksRepository.getTasks()).thenReturn(Observable.<List<Task>>empty());
+
+        //When subscribing to the statistics stream
+        mViewModel.getStatistics().subscribe(mStatisticsTestSubscriber);
+
+        //The correct pair is returned
+        Pair<Integer, Integer> result = mStatisticsTestSubscriber.getOnNextEvents().get(0);
+        assertThat(result.first, is(0));
+        assertThat(result.second, is(0));
+    }
+
+    @Test
+    public void getProgressIndicator_emits_afterStatisticsAreRetrieved() {
+        //Given a list of tasks in the repository
+        when(mTasksRepository.getTasks()).thenReturn(Observable.just(mTasks));
+
+        // And when subscribing to the progress indicator
+        mViewModel.getProgressIndicator().subscribe(mProgressIndicatorTestSubscriber);
+        //When subscribing to the statistics stream
+        mViewModel.getStatistics().subscribe();
+
+        // The intial value, false is emitted,
+        // then values true and false were emitted
+        mProgressIndicatorTestSubscriber.assertValues(false, true, false);
+    }
+
+    @Test
+    public void getProgressIndicator_emits_afterStatisticsAreRetrieved_WithError() {
+        //Given a list of tasks in the repository
+        when(mTasksRepository.getTasks()).thenReturn(Observable.<List<Task>>error(new Exception()));
+
+        // And when subscribing to the progress indicator
+        mViewModel.getProgressIndicator().subscribe(mProgressIndicatorTestSubscriber);
+        //When subscribing to the statistics stream
+        mViewModel.getStatistics().subscribe(mStatisticsTestSubscriber);
```

Comment:
Yes. `mViewModel.getStatistics()` is the one that has an error. Without a subscriber, the test will fail with `OnErrorNotImplementedException`